### PR TITLE
Add docker setup session for neuralchat finetuning sample

### PR DESCRIPTION
**Pull request by louie-tsai**
## Type of Change

Add docker setup session for neuralchat finetuning sample

## Description

add a docker setup session for customers

## Expected Behavior & Potential Risk

customers should be able to setup docker instance for this sample

## How has this PR been tested?

validate on AWS SPR instance

## Dependency Change?

no

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md [hunk 1]
```diff
@@ -14,12 +14,35 @@ This example demonstrates how to finetune the pretrained large language model (L
 # Prerequisite​
 
 ## 1. Environment​
+### Bare Metal
 Recommend python 3.9 or higher version.
 ```shell
 pip install -r requirements.txt
 # To use ccl as the distributed backend in distributed training on CPU requires to install below requirement.
 python -m pip install oneccl_bind_pt==1.13 -f https://developer.intel.com/ipex-whl-stable-cpu
```
Comment (alexsin368) [inline intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:22]: This command isn't working for me. Tried 1.13.0 and 1.13.0+cpu but also doesn't work.

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md [hunk 2]
```diff
@@ -14,12 +14,35 @@ This example demonstrates how to finetune the pretrained large language model (L
...
 ```
+### Docker 
+Pick either one of below options to setup docker environment.
+#### Option 1 : Build Docker image from scratch
+Please refer to this section : [How to build docker images for NeuralChat FineTuning](https://github.com/intel/intel-extension-for-transformers/tree/main/intel_extension_for_transformers/neural_chat/docker/finetuning#4-build-docker-image) to build docker image from scratch.  
+Once you have the docker image ready, please follow [run docker image](https://github.com/intel/intel-extension-for-transformers/tree/main/intel_extension_for_transformers/neural_chat/docker/finetuning#5-create-docker-container) session to launch a docker instance from the image.  
+
+#### Option 2: Pull existed Docker image
```
Comment (alexsin368) [inline intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:30]: existing

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md [hunk 3]
```diff
@@ -14,12 +14,35 @@ This example demonstrates how to finetune the pretrained large language model (L
...
+Please follow the session [itrex docker setup](https://github.com/intel/intel-extension-for-transformers/tree/main/docker#set-up-docker-image) and use the docker pull command to pull itrex docker image.  
+Once you have itrex docker image, follow below section to update itrex docker instance for this finetuning example.  
+```shell
+wget https://raw.githubusercontent.com/oneapi-src/oneAPI-samples/master/AI-and-Analytics/Getting-Started-Samples/IntelAIKitContainer_GettingStarted/run_oneapi_docker.sh
+cp requirement.txt /tmp
```
Comment (alexsin368) [inline intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:35]: should add an instruction to create tmp folder:
_mkdir tmp_

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md [hunk 4]
```diff
@@ -14,12 +14,35 @@ This example demonstrates how to finetune the pretrained large language model (L
...
+# change intel/ai-tools:itrex-0.1.1 according to itrex docker setup session
+./run_oneapi_docker.sh intel/ai-tools:itrex-0.1.1
+# don't need ipex in this sample
+pip uninstall pip install intel_extension_for_pytorch
```
Comment (alexsin368) [inline intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:39]: If you're just uninstalling IPEX, the command should be:
__pip uninstall pip install intel_extension_for_pytorch__

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md (hunk 1 above)
Comment (louie-tsai) [reply intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:22]: I don't change oneccl command. it also failed on me. I need to download it and install the wheel instead. need to create a JIRA accordingly.

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md (hunk 4 above)
Comment (louie-tsai) [reply intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:39]: oh. some duplication there. thanks for pointing that out.

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md (hunk 3 above)
Comment (louie-tsai) [reply intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:35]: /tmp is the system folder mount as ram disk.

**Review by lvliang-intel — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md (hunk 1 above)
Comment (hshen14) [reply intel_extension_for_transformers/neural_chat/examples/instruction_tuning/README.md:22]: is it a compatibility issue between torch and ccl?

**Pull request merged**